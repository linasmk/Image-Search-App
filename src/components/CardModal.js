/* ========= App Dependencies ============= */
import React from "react";
import PropTypes from "prop-types";
import { VscChromeClose } from "react-icons/vsc";
import Modal from "react-modal";
/* ========= Code ============= */

export const CardModal = ({
  openCardModal,
  closeCardModal,
  altDescription,
  imageID,
  likes,
  urlsFull,
  firstName,
  lastName,
  download,
}) => {
  const capitalizeFirstLetter = (string) =>
    string.charAt(0).toUpperCase() + string.slice(1);

  return (
    <Modal
      isOpen={openCardModal}
      onRequestClose={closeCardModal}
      contentLabel={altDescription}
      closeTimeoutMS={800}
      className="card-modal"
      key={imageID}
      style={{
        overlay: {
          backgroundColor: "rgba(0, 0, 0, .75)",
        },
        content: {
          color: "lightsteelblue",
        },
      }}
      ariaHideApp={false}>
      <div className="card-modal__content--wrapper">
        <VscChromeClose
          className="card-modal__close-btn"
          onClick={closeCardModal}
        />
        <article className="card-modal__content">
          <div className="card-modal__topbar" />
          <section className="card-modal__description-wrapper">
            <h2 className="card-modal__description">
              {altDescription != null
                ? capitalizeFirstLetter(altDescription)
                : ""}
            </h2>
          </section>
          <section className="card-modal__main">
            <img
              className="card-modal__img"
              alt={altDescription}
              src={urlsFull}
            />
          </section>
          <section className="card-modal__info-bar">
            <p className="card-modal__data">
              {likes != null ? `${likes} ` : 0}
              <span className="card-modal__text">Likes</span>
            </p>
            <p className="card-modal__data">
              <span className="card-modal__text">by </span>
              {firstName !== null ? firstName : `Uknown`}
              <span> </span>
              {lastName !== null ? lastName : `Uknown`}
            </p>
          </section>
          <section className="card-modal__btn-bar">
            <a
              className="card-modal__btn--download"
              href={`${download}?force=true`}
              download
              rel="noreferrer"
              target="_blank">
              Download
            </a>
          </section>
        </article>
      </div>
    </Modal>
  );
};
CardModal.defaultProps = {
  firstName: null,
  lastName: null,
  altDescription: null,
  likes: 0,
};
CardModal.propTypes = {
  openCardModal: PropTypes.bool.isRequired,
  closeCardModal: PropTypes.func.isRequired,
  imageID: PropTypes.string.isRequired,
  urlsFull: PropTypes.string.isRequired,
  download: PropTypes.string.isRequired,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  altDescription: PropTypes.string,
  likes: PropTypes.number,
};
export default CardModal;
