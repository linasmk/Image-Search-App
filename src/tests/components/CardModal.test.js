/* ========= App Dependencies ============= */
import React from "react";
import { shallow } from "enzyme";
/* ========= Components ============= */
import CardModal from "@comp/CardModal";
/* ========= Code ============= */
let wrapper;

beforeEach(() => {
  wrapper = shallow(<CardModal />);
});

it("renders CardModal component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
