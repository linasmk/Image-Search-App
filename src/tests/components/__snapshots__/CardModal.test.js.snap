// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardModal component correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="card-modal"
  closeTimeoutMS={800}
  contentLabel={null}
  isOpen={false}
  key="null"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "color": "lightsteelblue",
      },
      "overlay": Object {
        "backgroundColor": "rgba(0, 0, 0, .75)",
      },
    }
  }
>
  <div
    className="card-modal__content--wrapper"
  >
    <VscChromeClose
      className="card-modal__close-btn"
    />
    <article
      className="card-modal__content"
    >
      <div
        className="card-modal__topbar"
      />
      <section
        className="card-modal__description-wrapper"
      >
        <h2
          className="card-modal__description"
        />
      </section>
      <section
        className="card-modal__main"
      >
        <img
          alt={null}
          className="card-modal__img"
          src={null}
        />
      </section>
      <section
        className="card-modal__info-bar"
      >
        <p
          className="card-modal__data"
        >
          0 
          <span
            className="card-modal__text"
          >
            Likes
          </span>
        </p>
        <p
          className="card-modal__data"
        >
          <span
            className="card-modal__text"
          >
            by 
          </span>
          Uknown
          <span>
             
          </span>
          Uknown
        </p>
      </section>
      <section
        className="card-modal__btn-bar"
      >
        <a
          className="card-modal__btn--download"
          download={true}
          href="null?force=true"
          rel="noreferrer"
          target="_blank"
        >
          Download
        </a>
      </section>
    </article>
  </div>
</Modal>
`;
